AEA-1242 Update Q&A section and ledger section in docs (#2193)
fetchai/agents-aea: docs/questions-and-answers.md

<details><summary>What is an AEA?</summary>

> AEA stands for Autonomous Economic Agent. AEAs act independently of constant user input and autonomously execute actions to achieve their objective. Their goal is to create economic value for you, their owner.

Reviewer: (Bring consistent with the above)
```suggestion
AEA stands for Autonomous Economic Agent. AEAs act independently of constant user input and autonomously execute actions to achieve their goals. Their purpose is to create economic value for you, their owner.
```

<br><br>

> You can read more about an introduction to AEAs in the introduction.

Reviewer: ```suggestion
You can read an introduction to AEAs in <a>list<\a>.
```

Reviewer: I suggest:

`You can read an introduction to AEAs in <a>link<\a>.`

> You can read more about an introduction to AEAs in the introduction.

Reviewer: I suggest:
`You can read an introduction to AEAs in <a>link<\a>.`

Author: There is an issue with this link working with our link checker. Not worth the effort.

</details>

<details><summary>How do AEAs talk to each other when they do not know each other?</summary>

> For Autonomous Economic Agents (AEAs) to be able to talk to other AEAs, firstly they need to find them, then they need to use the same protocol for communication as them and finally they need to deliver messages to each other.

Reviewer: ```suggestion
For an Autonomous Economic Agent (AEA) to be able to talk to other AEAs, it first needs to find them. Once it does, it needs to use the same protocol for communication. If so, it then needs to send messages to them.
```

<br><br>
You can read more about Search and Discovery <a href="../oef-ledger/">here</a>, about protocols <a href="../core-components-1/">here</a> and about the Agent Communication Network <a href="../acn/">here</a>.
</details>

<details><summary>How does an AEA use blockchain?</summary>

> The AEA framework enables the agents to interact with public blockchains to complete transactions. Currently, the framework supports three different networks natively: the <i>Fetch.ai</i> network, the <i>Ethereum</i> network and the <i>Cosmos</i> network.

Reviewer: ```suggestion
The AEA framework enables the agents to interact with public blockchains to settle transactions. Currently, the framework has native support for three different networks: <i>Fetch.ai</i>, <i>Ethereum</i> and <i>Cosmos</i>.
```

</details>

<details><summary>How does one connect to a database?</summary>
You have two options to connect to a database: using the built-in storage solution or using a custom ORM (object-relational mapping) library and backend.
<br><br>
The use of the built-in storage is explained <a href="../generic-storage/">here</a>. For a detailed example of how to use an ORM follow the <a href="../orm-integration/">ORM guide</a>.
</details>

<details><summary>How does one connect a frontend?</summary>
There are multiple options, with the most obvious being the usage of a HTTP server connection and creation of a client that communicates with this connection.
<br><br>
You can find a more detailed discussion <a href="../connect-a-frontend/">here</a>.
</details>

<details><summary>Is the AEA framework ideal for agent-based modelling?</summary>

> The goal of agent-based modelling is to search for explanatory insight into the collective behaviour of agents obeying simple rules, typically in natural systems rather than in designing agents or solving specific practical or engineering problems.

Reviewer: ```suggestion
The goal of agent-based modelling (ABM) is to study the unknown (often complex) behaviour of systems comprised of agents with known (much simpler) behaviour. ABM is a popular technique in studying biological and social systems. Although there are some similarities between ABM and the AEA framework, the two have fundamentally different goals. ABM's goal is not the design of agents or solving specific practical or engineering problems.
```

Although it would be potentially possible, it would likely be inefficient to use the AEA framework for that kind of problem.
<br><br>
You can find more details on the application areas of the AEA framework <a href="../app-areas/">here</a>.
</details>

<details><summary>When a new AEA is created, is the <code>vendor</code> folder populated with some default packages?</summary>

> All AEA projects by default hold the <code>fetchai/stub:0.15.0</code> connection, the <code>fetchai/default:0.11.0</code>, <code>fetchai/state_update:0.9.0</code> and <code>fetchai/signing:0.9.0</code> protocol and the <code>fetchai/error:0.11.0</code> skill. These (as all other packages installed from the registry) are placed in the <code>vendor</code> folder.

Reviewer: ```suggestion
All AEA projects by default hold the <code>fetchai/stub:0.15.0</code> connection, the <code>fetchai/default:0.11.0</code>, <code>fetchai/state_update:0.9.0</code> and <code>fetchai/signing:0.9.0</code> protocols and the <code>fetchai/error:0.11.0</code> skill. These (as all other packages installed from the registry) are placed in the <code>vendor</code> folder.
```
